src/client/filter.rs

```diff
@@ -0,0 +1,292 @@
+/*-
+ * #%L
+ * OBKV Table Client Framework
+ * %%
+ * Copyright (C) 2024 OceanBase
+ * %%
+ * OBKV Table Client Framework is licensed under Mulan PSL v2.
+ * You can use this software according to the terms and conditions of the
+ * Mulan PSL v2. You may obtain a copy of Mulan PSL v2 at:
+ *          http://license.coscl.org.cn/MulanPSL2
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY
+ * KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ * See the Mulan PSL v2 for more details.
+ * #L%
+ */
+
+use std::{any::Any, fmt::Write};
+
+const TABLE_COMPARE_FILTER_PREFIX: &str = "TableCompareFilter";
+
+pub trait Filter: Any {
```

Comment:
In the rust, there are two ways to achieve the polymorphism:
- trait object
- enum

And here I find the `as_any` is used for downcasting, and it's not a good pattern when using trait object. And if the details of the underlying implementation are necessary, `enum` may be a good choice:
```
pub enum Filter {
     Value(ObTableValueFilter),
     List(ObTableFilterList),
}
```

Reply:
Fixed